src/gasPriceOracle/types.ts

```diff
@@ -17,3 +17,13 @@ export type SvmGasPriceEstimate = {
 export interface GasPriceFeed {
   (provider: PublicClient<Transport, Chain>, chainId: number): Promise<InternalGasPriceEstimate>;
 }
+
+export function isEVMGasPrice(gasPrice: GasPriceEstimate): gasPrice is EvmGasPriceEstimate {
+  const { maxFeePerGas, maxPriorityFeePerGas } = gasPrice as EvmGasPriceEstimate;
+  return isDefined(maxFeePerGas) && isDefined(maxPriorityFeePerGas);
+}
+
+export function isSVMGasPrice(gasPrice: GasPriceEstimate): gasPrice is SvmGasPriceEstimate {
+  const { baseFee, microLamportsPerComputeUnit } = gasPrice as SvmGasPriceEstimate;
+  return isDefined(baseFee) && isDefined(microLamportsPerComputeUnit);
+}
```

Comment:
nit: I wonder if in practice it might be more convenient to accept args of type `unknown` in these functions?

Reply:
Updated out of band - @gsteenkamp89 nice work on this; I really like this approach. It'll be interesting to see if we have other areas we can apply it.